<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>DESCRIPTION • FBCanalysis</title><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="DESCRIPTION"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">FBCanalysis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/Vignette.html">Develop and evaluate biomedical time series data clustering models based on Fluctuation Based Clustering (FBC)</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MrMaximumMax/FBCanalysis/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-title-body">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>DESCRIPTION</h1>
      <small class="dont-index">Source: <a href="https://github.com/MrMaximumMax/FBCanalysis/blob/HEAD/Reference_Manual_FBCanalysis.md" class="external-link"><code>Reference_Manual_FBCanalysis.md</code></a></small>
    </div>


<p>März 28, 2022</p>
<div id="description" class="section level1">

<pre><code>Package: FBCanalysis
Title: Develop and evaluate time series data models based on fluctuation based clustering
Version: 0.0.0.9000
Date: 2020-03-16
Authors@R: 
    person("Maximilian", "Koehler", , "maximilian.a.koehler@gmail.com", role = c("aut", "cre"))
Description: The package includes tools for performing fluctuation-based 
    clustering (FBC) on biological time series data, primarily for monitoring 
    data fluctuations in asthmatics. The package includes functions for 
    registering and processing time series data, developing matching clustering 
    models based on Earth Mover's distances, and evaluating the models through 
    enrichment analysis, stability after random data removal, or other frequently 
    used cluster stability metrics.
License: MIT + file LICENSE
Encoding: UTF-8
Roxygen: list(markdown = TRUE)
RoxygenNote: 7.1.2
URL: https://github.com/MrMaximumMax/FBCanalysis
BugReports: https://github.com/MrMaximumMax/FBCanalysis/issues
Depends:
    R (&gt;= 4.1.1)
Imports: 
    arsenal,
    cluster,
    clValid,
    dplyr,
    emdist,
    FCPS,
    glmnet,
    imputeTS,
    lubridate,
    mclust,
    RankAggreg,
    readr,
    stats,
    tibble,
    utils
Suggests: 
    rmarkdown,
    knitr
VignetteBuilder: knitr```


# `add_clust2enrich`

Add clustering assignments from clustering output to preprocessed enrichment
 data frame.


## Description

Add clustering assignments from clustering output to preprocessed enrichment
 data frame.


## Usage

```r
add_clust2enrich(enrich, clustdat)</code></pre>
<div class="section level2">
<h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>enrich</code></td>
<td>Preprocessed enrichment data frame (also see function: <a href="#addenrich">add_enrich</a> )</td>
</tr><tr class="even"><td><code>clustdat</code></td>
<td>Object of type list storing clustering data (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
<p>Processed enrichment data frame with added column indicating cluster assignments</p>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a></h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">clustering</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/clust_matrix.html">clust_matrix</a></span><span class="op">(</span><span class="va">matrix</span>, method <span class="op">=</span> <span class="st">"kmeans"</span>, nclust <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">enr</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_enrich.html">add_enrich</a></span><span class="op">(</span><span class="va">list</span>,
<span class="st">'https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/enrich/enrichment.csv'</span><span class="op">)</span>
<span class="va">enr</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_clust2enrich.html">add_clust2enrich</a></span><span class="op">(</span><span class="va">enr</span>, <span class="va">clustering</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="add_clust2ts"><code>add_clust2ts</code><a class="anchor" aria-label="anchor" href="#add_clust2ts"></a></h1>
<p>Add clustering assignments from clustering output to time series data list and store the data in a data frame.</p>
<div class="section level2">
<h2 id="description-1">Description<a class="anchor" aria-label="anchor" href="#description-1"></a></h2>
<p>Add clustering assignments from clustering output to time series data list and store the data in a data frame.</p>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a></h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/add_clust2ts.html">add_clust2ts</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">clustdat</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-1">Arguments<a class="anchor" aria-label="anchor" href="#arguments-1"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>clustdat</code></td>
<td>Object of type list storing clustering data (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-1">Value<a class="anchor" aria-label="anchor" href="#value-1"></a></h2>
<p>Processed data frame storing time series data with added column indicating cluster assignments</p>
</div>
<div class="section level2">
<h2 id="examples-1">Examples<a class="anchor" aria-label="anchor" href="#examples-1"></a></h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">clustering</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/clust_matrix.html">clust_matrix</a></span><span class="op">(</span><span class="va">matrix</span>, method <span class="op">=</span> <span class="st">"kmeans"</span>, nclust <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">ts</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_clust2ts.html">add_clust2ts</a></span><span class="op">(</span><span class="va">list</span>, <span class="va">clustering</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="add_enrich"><code>add_enrich</code><a class="anchor" aria-label="anchor" href="#add_enrich"></a></h1>
<p>Add enrichment data from a csv file, match with Patient ID entries from a previously generated time series data list and preprocess for further analysis.</p>
<div class="section level2">
<h2 id="description-2">Description<a class="anchor" aria-label="anchor" href="#description-2"></a></h2>
<p>Add enrichment data from a csv file, match with Patient ID entries from a previously generated time series data list and preprocess for further analysis.</p>
</div>
<div class="section level2">
<h2 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a></h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/add_enrich.html">add_enrich</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">path</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-2">Arguments<a class="anchor" aria-label="anchor" href="#arguments-2"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>path</code></td>
<td>Path where enrichment csv file is stored</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
<p>the enrichment csv file should have a column including the Patient ID. Additionally, the user specifies the list in which time series data is saved. This is advantageous since the function can now do matching, i.e. determine which Patient IDs occur in both the enrichment dataset and time series datalist. So for a result, any Patient ID that appears in the enrichment dataset but does not exist in the time series datalist will be deleted from the enrichment dataset, as it cannot be used in any further investigation. Nonetheless, Patient IDs from the time series data that do not appear in the enrichment dataset will be added to the enrichment dataset, but each new parameter will be featureless, so added as NA value.</p>
<p>If the user selects option 1 (leave missing values as NA), no further processing of the input occurs. The enrichment data set will be added to the environment as a data frame. In this situation, the NA values from the enrichment dataset will be included in the summary indicating, for example, that a certain cluster has a given percentage of missing values. This may also lead to some additional findings, such as that a specific parameter considerably enriches a cluster yet many data is absent. If the user selects option 2 (sample missing values), the function loops over each NA entry and selects a random value from the whole distribution for the parameter for which the data is missing. This cycle is repeated until the whole dataset has been processed and the data will be added as a data frame to the environment.</p>
</div>
<div class="section level2">
<h2 id="value-2">Value<a class="anchor" aria-label="anchor" href="#value-2"></a></h2>
<p>Processed data as object of type data frame; Enrichment data Patient_IDs are matched with Time Series Data List Patient IDs; In case it was indicated, NA values in the enrichment data are filled up by random sampling</p>
</div>
<div class="section level2">
<h2 id="examples-2">Examples<a class="anchor" aria-label="anchor" href="#examples-2"></a></h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">enr</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_enrich.html">add_enrich</a></span><span class="op">(</span><span class="va">list</span>,
<span class="st">'https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/enrich/enrichment.csv'</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="clust_matrix"><code>clust_matrix</code><a class="anchor" aria-label="anchor" href="#clust_matrix"></a></h1>
<p>Cluster Earth Mover Distance Square Matrix data and record cluster assignments for involved Patient_IDs for a specified clustering technique and number of clusters.</p>
<div class="section level2">
<h2 id="description-3">Description<a class="anchor" aria-label="anchor" href="#description-3"></a></h2>
<p>Cluster Earth Mover Distance Square Matrix data and record cluster assignments for involved Patient_IDs for a specified clustering technique and number of clusters.</p>
</div>
<div class="section level2">
<h2 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a></h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/clust_matrix.html">clust_matrix</a></span><span class="op">(</span><span class="va">matrix</span>, <span class="va">method</span>, <span class="va">nclust</span>, <span class="va">plotclust</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-3">Arguments<a class="anchor" aria-label="anchor" href="#arguments-3"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>matrix</code></td>
<td>Object of type matrix storing Earth Mover’s Distances for patient time series data distribution pairs</td>
</tr><tr class="even"><td><code>method</code></td>
<td>Clustering method (hierarchical, kmeans, diana, fanny, som, modelbased, sota, pam, clara)</td>
</tr><tr class="odd"><td><code>nclust</code></td>
<td>Number of clusters (if not specified, user will be asked in the terminal)</td>
</tr><tr class="even"><td><code>plotclust</code></td>
<td>TRUE/FALSE if clustering data should be visualized (TRUE by default)</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-1">Details<a class="anchor" aria-label="anchor" href="#details-1"></a></h2>
<p>Hierarchical clustering describes a general agglomerative hierarchical clustering approach in which the optimum value of an objective function is used to choose which pair of clusters should merge at each step (see <a href="#hclust">hclust</a> for more details).</p>
<p>K-means clustering is a vector quantization technique that divides a set of n observations into k groups, with each observation belonging to the cluster with the closest mean or centroid (see <a href="#kmeans">kmeans</a> for more details).</p>
<p>The divisive analytic method (DIANA) constructs a hierarchical clustering structure, starting with a single huge cluster containing all n observations. Clusters are further split until each has a single observation. At each step, the cluster with the largest diameter is selected, where the diameter of a cluster is defined as the biggest difference between any two of its observations (see <a href="#diana">diana</a> for more details).</p>
<p>While partitioning around medoids (PAM) is comparable to k-means, it is considered more robust since it allows for the use of dissimilarities other than euclidean distance. As with k-means, the number of clusters is determined in advance, and an initial set of cluster centers is required to begin the process (see <a href="#pam">pam</a> for more details).</p>
<p>Clustering large applications (CLARA) is a system that involves sampling to apply PAM to a sequence of sub-datasets. When the number of observations is big, this results in shorter run times. It is substantially faster than other partitioning algorithms such as PAM at handling huge datasets. Internally, this is performed by examining fixed-size sub-datasets, which results in linear rather than quadratic time and storage requirements for n (see <a href="#clara">clara</a> for more details).</p>
<p>FANNY explains a fuzzy analysis clustering method. Each observation is distributed through-out the numerous groups in a fuzzy clustering (see <a href="#fanny">fanny</a> for more details).</p>
<p>Self-organizing maps (SOM) are a widespread unsupervised learning technique used by com- putational biologists and academics in machine learning. SOM is a neural network-based system that is well-known for its ability to map and display two-dimensional data (see <a href="#somclustering">SOMclustering</a> for more details).</p>
<p>Modelbased clusterinng fits the data to a statistical model composed of a finite mixture of Gaussian distributions. Each mixture component represents a cluster, and the maximum like- lihood method, or estimation maximum (EM), is used to determine the mixture components and group memberships (see <a href="#modelbasedclustering">ModelBasedClustering</a> for more details).</p>
<p>SOTA, self-organizing tree algorithm, denotes an unsupervised network with a hierarchical and divisive binary tree topology. It is a fast approach, which makes it suitable for clustering a large number of elements. It combines the advantages of hierarchical clustering with those of SOMs. The algorithm chooses the most diverse node and separates it into two nodes referred as cells (see <a href="#sota">sota</a> for more details).</p>
</div>
<div class="section level2">
<h2 id="value-3">Value<a class="anchor" aria-label="anchor" href="#value-3"></a></h2>
<p>Object of type list storing cluster data and clustering assignments for the Patient_IDs from the Earth Mover’s Distance matrix</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
<p>Joe H Ward Jr. Hierarchical grouping to optimize an objective function. Journal of the American statistical association, 58(301):236–244, 1963.</p>
<p>Stephane Tuff ́ery. Data mining and statistics for decision making. John Wiley &amp; Sons, 2011.</p>
<p>Fionn Murtagh. A survey of recent advances in hierarchical clustering algorithms. The computer journal, 26(4):354–359, 1983.</p>
<p>Fionn Murtagh. Clustering in massive data sets. In Handbook of massive data sets, pages 501–543. Springer, 2002.</p>
<p>Greg Hamerly and Charles Elkan. Alternatives to the k-means algorithm that find better clusterings. In Proceedings of the eleventh international conference on Information and knowledge management, pages 600–607, 2002.</p>
<p>R Wehrens and J Kruisselbrink. kohonen: Supervised and unsupervised self-organising maps r package version 3.0. 10, 2019.</p>
<p>Leonard Kaufman and Peter J Rousseeuw. Finding groups in data: an introduction to cluster analysis. John Wiley &amp; Sons, 2009.</p>
<p>Mark Van der Laan, Katherine Pollard, and Jennifer Bryan. A new partitioning around medoids algorithm. Journal of Statistical Computation and Simulation, 73(8):575–584, 2003.</p>
<p>Chris Fraley and Adrian E Raftery. Model-based clustering, discriminant analysis, and density estimation. Journal of the American statistical Association, 97(458):611–631, 2002.</p>
<p>Javier Herrero, Alfonso Valencia, and Joaquın Dopazo. A hierarchical unsupervised growing neural network for clustering gene expression patterns. Bioinformatics, 17(2):126–136, 2001.</p>
</div>
<div class="section level2">
<h2 id="examples-3">Examples<a class="anchor" aria-label="anchor" href="#examples-3"></a></h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/emd_matrix.html">emd_matrix</a></span><span class="op">(</span><span class="va">list</span>, <span class="st">"FEV1"</span><span class="op">)</span>
<span class="va">clustering</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/clust_matrix.html">clust_matrix</a></span><span class="op">(</span><span class="va">matrix</span>, method <span class="op">=</span> <span class="st">"hierarchical"</span>, nclust <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="clvalid_flow"><code>clValid_flow</code><a class="anchor" aria-label="anchor" href="#clvalid_flow"></a></h1>
<p>Interactive console workflow to calculate and evaluate cluster validation measures which have been determined previously by the call <a href="#initclvalid">init_clValid</a> . The call guides through internal cluster measures, cluster stability measure where the results of the measured will be presented and summarized in the console. Finally, a summary and optimized rank for the measures on the inidcated clustering techniques can be shown and a Cross Entropy Monte Carlo Search Algorithm can choose the ideal clustering model by applyinng Spearman’s footrule distance.</p>
<div class="section level2">
<h2 id="description-4">Description<a class="anchor" aria-label="anchor" href="#description-4"></a></h2>
<p>Interactive console workflow to calculate and evaluate cluster validation measures which have been determined previously by the call <a href="#initclvalid">init_clValid</a> . The call guides through internal cluster measures, cluster stability measure where the results of the measured will be presented and summarized in the console. Finally, a summary and optimized rank for the measures on the inidcated clustering techniques can be shown and a Cross Entropy Monte Carlo Search Algorithm can choose the ideal clustering model by applyinng Spearman’s footrule distance.</p>
</div>
<div class="section level2">
<h2 id="usage-3">Usage<a class="anchor" aria-label="anchor" href="#usage-3"></a></h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/clValid_flow.html">clValid_flow</a></span><span class="op">(</span><span class="va">matrix</span>, <span class="va">par</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-4">Arguments<a class="anchor" aria-label="anchor" href="#arguments-4"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>matrix</code></td>
<td>Earth Mover’s Distance Matrix for processed patient time series data (also see functions: <a href="#emdmatrix">emd_matrix</a> , <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>par</code></td>
<td>Object of type list storing clustering methods and cluster range of interest; initialized via function: <a href="#initclvalid">init_clValid</a>
</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-2">Details<a class="anchor" aria-label="anchor" href="#details-2"></a></h2>
<p>The call guides through an interactive workflow and generates cluster evaluation measures, stores and lists, visualizes corresponding plots and lets the user decide which technique is the prefered one. Once the user has chosen his favourite, the flow continues to the function <a href="#clustmatrix">clust_matrix</a> and generates the respective clustering output. The internal cluster validation methods utilize just the dataset and the clustering partition as input and evaluates the clustering’s quality by using intrinsic information included in the data.</p>
<p>The call calculates Connectivity, Silhouette width and Dunn index. Connectivity describes the connectness to neighbors of particular clustering partition and should be minimized. Silhouette width defines the average silhouette value for each observation and should be maximized. The Dunnn index is a definition for Ratio of shortest distance between non-cluster observation and greatest intra-cluster distance and should be maximized likewise.</p>
<p>Furthermore, cluster stability measures are available, namely Average proportion of non-overlap (APN), Average distance (AD), Average distance between means (ADM) and Figure of merit (FOM). APN is the average proportion of observations that are not clustered using complete and leaky data. AD defines the average distance in observations for both complete and leaky data. ADM deals with the average distance between cluster centers in complete and leaky data. FOM is a measure for average intra- cluster variance in leaky data. All measures should be minimized. Furthermore, Rank Aggregation may be performed. It approaches to provide a generic and flexible framework for objectively integrating several ordered lists in a suitable and efficient way. The used technique for evaluating clustering the rank is by a Cross-entropy approach, which is incorporating Spearman’s footrule distance measure. In the end a recommendation for the best fitting clustering model is given.</p>
</div>
<div class="section level2">
<h2 id="value-4">Value<a class="anchor" aria-label="anchor" href="#value-4"></a></h2>
<p>Object of type list storing chosen clustering method and number of clusters (can be then used for function <a href="#clustmatrix">clust_matrix</a> )</p>
</div>
<div class="section level2">
<h2 id="references-1">References<a class="anchor" aria-label="anchor" href="#references-1"></a></h2>
<p>Guy Brock, Vasyl Pihur, Susmita Datta, and Somnath Datta. clvalid: An r package for cluster validation. Journal of Statistical Software, 25:1–22, 2008.</p>
<p>Julia Handl, Joshua Knowles, and Douglas B Kell. Computational cluster validation in post-genomic data analysis. Bioinformatics, 21(15):3201–3212, 2005.</p>
<p>Peter J Rousseeuw. Silhouettes: a graphical aid to the interpretation and validation of cluster analysis. Journal of computational and applied mathematics, 20:53–65, 1987.</p>
<p>Joseph C Dunn. Well-separated clusters and optimal fuzzy partitions. Journal of cybernetics, 4(1):95–104, 1974.</p>
<p>Vasyl Pihur, Susmita Datta, and Somnath Datta. Weighted rank aggregation of cluster validation measures: a monte carlo cross-entropy approach. Bioinformatics, 23(13):1607–1615, 2007.</p>
</div>
<div class="section level2">
<h2 id="examples-4">Examples<a class="anchor" aria-label="anchor" href="#examples-4"></a></h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">distmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/emd_matrix.html">emd_matrix</a></span><span class="op">(</span><span class="va">list</span>, <span class="st">"PEF"</span>, maxIter <span class="op">=</span> <span class="fl">5000</span><span class="op">)</span>
<span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/init_clValid.html">init_clValid</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/clValid_flow.html">clValid_flow</a></span><span class="op">(</span><span class="va">distmat</span>, <span class="va">parameters</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="emd_heatmap"><code>emd_heatmap</code><a class="anchor" aria-label="anchor" href="#emd_heatmap"></a></h1>
<p>Visualize an Earth Mover’s Distance Square Matrix as a heatmap</p>
<div class="section level2">
<h2 id="description-5">Description<a class="anchor" aria-label="anchor" href="#description-5"></a></h2>
<p>Visualize an Earth Mover’s Distance Square Matrix as a heatmap</p>
</div>
<div class="section level2">
<h2 id="usage-4">Usage<a class="anchor" aria-label="anchor" href="#usage-4"></a></h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/emd_heatmap.html">emd_heatmap</a></span><span class="op">(</span><span class="va">input</span>, <span class="va">parameter</span>, <span class="va">maxIter</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-5">Arguments<a class="anchor" aria-label="anchor" href="#arguments-5"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>input</code></td>
<td>Earth Mover’s Distance Matrix or list storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>In case list is input, the parameter of interest from time series data list</td>
</tr><tr class="odd"><td><code>Iter</code></td>
<td>In case input is time series data list, incate maxIter to calculate EMD matrix (also see function: <a href="#emdmatrix">emd_matrix</a> )</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-5">Value<a class="anchor" aria-label="anchor" href="#value-5"></a></h2>
<p>Visualized Earth Mover’s Distance Matrix as a heatmap</p>
</div>
<div class="section level2">
<h2 id="examples-5">Examples<a class="anchor" aria-label="anchor" href="#examples-5"></a></h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/emd_matrix.html">emd_matrix</a></span><span class="op">(</span><span class="va">list</span>, <span class="st">"FEV1"</span><span class="op">)</span>
<span class="fu"><a href="reference/emd_heatmap.html">emd_heatmap</a></span><span class="op">(</span><span class="va">matrix</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="emd_matrix"><code>emd_matrix</code><a class="anchor" aria-label="anchor" href="#emd_matrix"></a></h1>
<p>Generate an Earth Mover’s Distance Matrix for time series data distributions pairs out of a preprocessed time series data list.</p>
<div class="section level2">
<h2 id="description-6">Description<a class="anchor" aria-label="anchor" href="#description-6"></a></h2>
<p>Generate an Earth Mover’s Distance Matrix for time series data distributions pairs out of a preprocessed time series data list.</p>
</div>
<div class="section level2">
<h2 id="usage-5">Usage<a class="anchor" aria-label="anchor" href="#usage-5"></a></h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/emd_matrix.html">emd_matrix</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">parameter</span>, <span class="va">maxIter</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-6">Arguments<a class="anchor" aria-label="anchor" href="#arguments-6"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>Parameter of interest to determine Earth Mover’s Distances between distributions</td>
</tr><tr class="odd"><td><code>maxIter</code></td>
<td>Maximum of iterations to calculate Earth Mover’s Distance (default: 500)</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-3">Details<a class="anchor" aria-label="anchor" href="#details-3"></a></h2>
<p>the function may compute the EMD for each Patient ID pairi,j using the normalized distributions. EMD is a distance measure between two probability distributions over a region D. Informally, if the distributions are viewed as two distinct methods of accumulating a certain quantity of earth gravel across the region D. EMD is the smallest cost associated with converting one pile to another, where the cost is supposed to equal the quantity of material transferred multiplied by the distance traveled. A unit of labor is defined in this context as conveying a unit of earth across a unit of ground distance. A distribution may be described as a collection of clusters, each of which is defined by its mean or mode and the proportion of the distribution that belongs to it. This representation is referred to as the distribution’s signature. Both signatures may be of varying sizes. Simple distributions, for example, have lower signatures than complex distributions. (also see <a href="#emdist">emdist</a> for further details)</p>
</div>
<div class="section level2">
<h2 id="value-6">Value<a class="anchor" aria-label="anchor" href="#value-6"></a></h2>
<p>Earth Mover’s Distance Square Matrix of type matrix</p>
</div>
<div class="section level2">
<h2 id="references-2">References<a class="anchor" aria-label="anchor" href="#references-2"></a></h2>
<p>Yossi Rubner, Carlo Tomasi, and Leonidas J Guibas. A metric for distributions with applications to image databases. In Sixth International Conference on Computer Vi- sion (IEEE Cat. No. 98CH36271), pages 59–66. IEEE, 1998.</p>
</div>
<div class="section level2">
<h2 id="examples-6">Examples<a class="anchor" aria-label="anchor" href="#examples-6"></a></h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/emd_matrix.html">emd_matrix</a></span><span class="op">(</span><span class="va">list</span>, <span class="st">"FEV1"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="enr_obs_clust"><code>enr_obs_clust</code><a class="anchor" aria-label="anchor" href="#enr_obs_clust"></a></h1>
<p>Observe a specific cluster of interest on preporcessed enrichment and time series data for overview and p-values. The p-values are calculated by a Wilcox Rank Sum test for continuous data and a Fisher’s Exact Test for categorical enrichment data.</p>
<div class="section level2">
<h2 id="description-7">Description<a class="anchor" aria-label="anchor" href="#description-7"></a></h2>
<p>Observe a specific cluster of interest on preporcessed enrichment and time series data for overview and p-values. The p-values are calculated by a Wilcox Rank Sum test for continuous data and a Fisher’s Exact Test for categorical enrichment data.</p>
</div>
<div class="section level2">
<h2 id="usage-6">Usage<a class="anchor" aria-label="anchor" href="#usage-6"></a></h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/enr_obs_clust.html">enr_obs_clust</a></span><span class="op">(</span><span class="va">ts.dat</span>, <span class="va">enrich</span>, <span class="va">clustno</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-7">Arguments<a class="anchor" aria-label="anchor" href="#arguments-7"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>ts.dat</code></td>
<td>Processed data frame storing time series data and cluster assignments (also see function: <a href="#addclust2ts">add_clust2ts</a> )</td>
</tr><tr class="even"><td><code>enrich</code></td>
<td>Processed data frame storing enrichment data and cluster assignments (also see function: <a href="#addclust2enrich">add_clust2enrich</a> )</td>
</tr><tr class="odd"><td><code>clustno</code></td>
<td>Cluster number of interest</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-4">Details<a class="anchor" aria-label="anchor" href="#details-4"></a></h2>
<p>There are two techniques to compute the p-value for continuous or categorical data, according to past work. In order to determine the relevant p-value inside a cluster of interest, the data distribution within the cluster should be compared to the data distribution outside the cluster. Prior to conducting the related probability tests, one data processing step is performed, namely the construction of two data distributions, one including only data included inside the cluster and another comprising data from outside the cluster, for the purpose of comparing them.</p>
<p>The Mann-Whitney Test, sometimes referred to as the Wilcoxon rank-sum test (WRS), is used to measure the significance of continuous variables within the observed distribution. The WRS is used to test if the central tendency of two independent samples is different. When the t-test for independent samples does not meet the requirements, the WRS is used. The null hypothesis H0 states that the populations’ distributions are equal. H1 is the alternative hypothesis meaning that the distributions are not equal. The test is consistent under the broader formulation only when the following happens under H1.</p>
<p>The hypergeometric test or Fisher’s exact test (FET) is used to analyse categorical variables within the enriched data set. It is a statistical significance test for contingency tables that is employed in the study of them. The test is helpful for categorical data derived from object classification. It is used to assess the importance of associations and inconsistencies between classes. The FET is often used in conjunction with a 2 × 2 contingency table that represents two categories for a variable, as well as assignment inside or outside of the cluster. The p-value is calculated as if the table’s margins are fixed. This results in a hypergeometric distribution of the numbers in the table cells under the null hypothesis of independence. A hypergeometric distribution is a discrete probability distribution that describes the probability of k successes, defined as random draws for which the object drawn has a specified feature in n draws without replacement from a finite population of size N containing exactly K objects with that feature, where each draw is either successful or unsuccessful. The test is only practicable for normal computations in the presence of a 2 × 2 contingency table. However, the test’s idea may be extended to the situation of a m × n table in general. Statistics programs provide a Monte Carlo approach for approximating the more general case.</p>
</div>
<div class="section level2">
<h2 id="value-7">Value<a class="anchor" aria-label="anchor" href="#value-7"></a></h2>
<p>Terminal output presenting summary of time series and enrichment data with corresponding p-values</p>
</div>
<div class="section level2">
<h2 id="references-3">References<a class="anchor" aria-label="anchor" href="#references-3"></a></h2>
<p>Siegel Sidney. Nonparametric statistics for the behavioral sciences. The Journal of Nervous and Mental Disease, 125(3):497, 1957.</p>
<p>Kinley Larntz. Small-sample comparisons of exact levels for chi-squared goodness-of-fit statistics. Journal of the American Statistical Association, 73(362):253–263, 1978.</p>
<p>Cyrus R Mehta and Nitin R Patel. A network algorithm for performing fisher’s exact test in r× c contingency tables. Journal of the American Statistical Association, 78(382):427–434, 1983.</p>
<p>Aravind Subramanian, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, Scott L Pomeroy, Todd R Golub, Eric S Lander, et al. Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. Proceedings of the National Academy of Sciences, 102(43):15545–15550, 2005.</p>
</div>
<div class="section level2">
<h2 id="examples-7">Examples<a class="anchor" aria-label="anchor" href="#examples-7"></a></h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">clustering</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/clust_matrix.html">clust_matrix</a></span><span class="op">(</span><span class="va">matrix</span>, method <span class="op">=</span> <span class="st">"kmeans"</span>, nclust <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">enr</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_enrich.html">add_enrich</a></span><span class="op">(</span><span class="va">list</span>,
<span class="st">'https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/enrich/enrichment.csv'</span><span class="op">)</span>
<span class="va">enr</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_clust2enrich.html">add_clust2enrich</a></span><span class="op">(</span><span class="va">enr</span>, <span class="va">clustering</span><span class="op">)</span>
<span class="va">ts</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/add_clust2ts.html">add_clust2ts</a></span><span class="op">(</span><span class="va">list</span>, <span class="va">clustering</span><span class="op">)</span>
<span class="fu"><a href="reference/enr_obs_clust.html">enr_obs_clust</a></span><span class="op">(</span><span class="va">ts</span>, <span class="va">enr</span>, <span class="fl">1</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="fbcanalysis"><code>FBCanalysis</code><a class="anchor" aria-label="anchor" href="#fbcanalysis"></a></h1>
<p>FBCanalysis: A package for developing and evaluating biomedical time series data clustering model based on Fluctuation Based Clustering (FBC)</p>
<div class="section level2">
<h2 id="description-8">Description<a class="anchor" aria-label="anchor" href="#description-8"></a></h2>
<p>This R package aims to offer researchers with fast tools for clustering patient time series data and confirming the distinction using additional metrics such as population parameter enrichment analysis, stability after random data removal, and conventional cluster stability measures. The package attempts to conveniently apply computational methods and capabilities for developing and evaluating unsupervised clustering models with the goal of data-drivenly categorizing asthmatic patients according to their illness dynamics.</p>
</div>
<div class="section level2">
<h2 id="details-5">Details<a class="anchor" aria-label="anchor" href="#details-5"></a></h2>
<p>clustering may be used within the proposed package to identify significant diverse groupings in a patient population, and enrichment analysis is used to examine any possible correlations with clinically relevant characteristics.</p>
<p>The R package thus aims to offer researchers with fast tools for clustering patient time series data and confirming the distinction using additional metrics such as population parameter enrichment analysis, stability after random data removal, and conventional cluster stability measures</p>
</div>
<div class="section level2">
<h2 id="seealso">Seealso<a class="anchor" aria-label="anchor" href="#seealso"></a></h2>
<p><a href="https://github.com/MrMaximumMax/FBCanalysis" class="external-link">GitHub Repository</a></p>
</div>
</div>
<div class="section level1">
<h1 id="init_clvalid"><code>init_clValid</code><a class="anchor" aria-label="anchor" href="#init_clvalid"></a></h1>
<p>Initialize Cluster Validation Measure Analysis in the context of Fluctuation Based Clustering (FBC) analysis. The call initialized a interactive console workflow where the user may indicate the clustering techniques of interest as well as the cluster numbers of interest.</p>
<div class="section level2">
<h2 id="description-9">Description<a class="anchor" aria-label="anchor" href="#description-9"></a></h2>
<p>Initialize Cluster Validation Measure Analysis in the context of Fluctuation Based Clustering (FBC) analysis. The call initialized a interactive console workflow where the user may indicate the clustering techniques of interest as well as the cluster numbers of interest.</p>
</div>
<div class="section level2">
<h2 id="usage-7">Usage<a class="anchor" aria-label="anchor" href="#usage-7"></a></h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/init_clValid.html">init_clValid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="details-6">Details<a class="anchor" aria-label="anchor" href="#details-6"></a></h2>
<p>See <a href="#clustmatrix">clust_matrix</a> for more details on possible clustering techniques</p>
</div>
<div class="section level2">
<h2 id="value-8">Value<a class="anchor" aria-label="anchor" href="#value-8"></a></h2>
<p>Object of type list storing cluster method(s) and number of cluster range of interest (to be used for function: <a href="#clvalidflow">clValid_flow</a> )</p>
</div>
<div class="section level2">
<h2 id="references-4">References<a class="anchor" aria-label="anchor" href="#references-4"></a></h2>
<p>Guy Brock, Vasyl Pihur, Susmita Datta, and Somnath Datta. clvalid: An r package for cluster validation. Journal of Statistical Software, 25:1–22, 2008.</p>
</div>
<div class="section level2">
<h2 id="examples-8">Examples<a class="anchor" aria-label="anchor" href="#examples-8"></a></h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/init_clValid.html">init_clValid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="jaccard_run_cognate"><code>jaccard_run_cognate</code><a class="anchor" aria-label="anchor" href="#jaccard_run_cognate"></a></h1>
<p>Simulate amount of random data removal from time series data list and determine Jaccard index via Cognate Cluster approach for multiple random data removal steps for a specific cluster of interest.</p>
<div class="section level2">
<h2 id="description-10">Description<a class="anchor" aria-label="anchor" href="#description-10"></a></h2>
<p>Simulate amount of random data removal from time series data list and determine Jaccard index via Cognate Cluster approach for multiple random data removal steps for a specific cluster of interest.</p>
</div>
<div class="section level2">
<h2 id="usage-8">Usage<a class="anchor" aria-label="anchor" href="#usage-8"></a></h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">jaccard_run_cognate</span><span class="op">(</span>
  <span class="va">plist</span>,
  <span class="va">parameter</span>,
  <span class="va">n_simu</span>,
  <span class="va">method</span>,
  <span class="va">clust_num</span>,
  <span class="va">n_clust</span>,
  <span class="va">range</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-8">Arguments<a class="anchor" aria-label="anchor" href="#arguments-8"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>Object of type list storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>Parameter of interest in time series data list</td>
</tr><tr class="odd"><td><code>n_simu</code></td>
<td>Number of simulations</td>
</tr><tr class="even"><td><code>method</code></td>
<td>Clustering method (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="odd"><td><code>clust_num</code></td>
<td>Cluster of interest</td>
</tr><tr class="even"><td><code>n_clust</code></td>
<td>Number of clusters</td>
</tr><tr class="odd"><td><code>range</code></td>
<td>Range to simulate random data removal (e.g. c(0.1,0.2,0.5,0.7,0.8))</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-7">Details<a class="anchor" aria-label="anchor" href="#details-7"></a></h2>
<p>See <a href="#simjaccardcognate">sim_jaccard_cognate</a> for more detailed approach on Jaccard index determination. The difference in this function is that now only one cluster is observed für multiple amoiunts of random data removal where for each data removal step defined the resulting Jaccard indices are stored in a list object. Furthermore, a boxplot visualization is generated, in the style of recent publications.</p>
</div>
<div class="section level2">
<h2 id="value-9">Value<a class="anchor" aria-label="anchor" href="#value-9"></a></h2>
<p>Object of type list storing Jaccard indices for each indicated random data removal step and visualized results in a boxplot</p>
</div>
<div class="section level2">
<h2 id="references-5">References<a class="anchor" aria-label="anchor" href="#references-5"></a></h2>
<p>Anja Jochmann, Luca Artusio, Hoda Sharifian, Angela Jamalzadeh, Louise J Fleming, Andrew Bush, Urs Frey, and Edgar Delgado-Eckert. Fluctuation-based clustering reveals phenotypes of patients with different asthma severity. ERJ open research, 6(2), 2020.</p>
</div>
<div class="section level2">
<h2 id="examples-9">Examples<a class="anchor" aria-label="anchor" href="#examples-9"></a></h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">output</span> <span class="op">&lt;-</span> <span class="fu">jaccard_run_cognate</span><span class="op">(</span><span class="va">list</span>,<span class="st">"PEF"</span>,<span class="fl">10</span>,<span class="st">"hierarchical"</span>,<span class="fl">1</span>,<span class="fl">3</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.005</span>,<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="jaccard_run_emd"><code>jaccard_run_emd</code><a class="anchor" aria-label="anchor" href="#jaccard_run_emd"></a></h1>
<p>Simulate amount of random data removal from time series data list and determine Jaccard index via Earth Mover’s Distance approach for multiple random data removal steps for a specific cluster of interest.</p>
<div class="section level2">
<h2 id="description-11">Description<a class="anchor" aria-label="anchor" href="#description-11"></a></h2>
<p>Simulate amount of random data removal from time series data list and determine Jaccard index via Earth Mover’s Distance approach for multiple random data removal steps for a specific cluster of interest.</p>
</div>
<div class="section level2">
<h2 id="usage-9">Usage<a class="anchor" aria-label="anchor" href="#usage-9"></a></h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/jaccard_run_emd.html">jaccard_run_emd</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">parameter</span>, <span class="va">n_simu</span>, <span class="va">method</span>, <span class="va">clust_num</span>, <span class="va">n_clust</span>, <span class="va">range</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-9">Arguments<a class="anchor" aria-label="anchor" href="#arguments-9"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>Object of type list storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>Parameter of interest in time series data list</td>
</tr><tr class="odd"><td><code>n_simu</code></td>
<td>Number of simulations</td>
</tr><tr class="even"><td><code>method</code></td>
<td>Clustering method (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="odd"><td><code>clust_num</code></td>
<td>Cluster of interest</td>
</tr><tr class="even"><td><code>n_clust</code></td>
<td>Number of clusters</td>
</tr><tr class="odd"><td><code>range</code></td>
<td>Range to simulate random data removal (e.g. c(0.1,0.2,0.5,0.7,0.8))</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-8">Details<a class="anchor" aria-label="anchor" href="#details-8"></a></h2>
<p>See <a href="#simjaccardemd">sim_jaccard_emd</a> for more detailed approach on Jaccard index determination. The difference in this function is that now only one cluster is observed für multiple amoiunts of random data removal where for each data removal step defined the resulting Jaccard indices are stored in a list object. Furthermore, a boxplot visualization is generated, in the style of recent publications.</p>
</div>
<div class="section level2">
<h2 id="value-10">Value<a class="anchor" aria-label="anchor" href="#value-10"></a></h2>
<p>Object of type list storing Jaccard indices for each indicated random data removal step and visualized results in a boxplot</p>
</div>
<div class="section level2">
<h2 id="references-6">References<a class="anchor" aria-label="anchor" href="#references-6"></a></h2>
<p>Yossi Rubner, Carlo Tomasi, and Leonidas J Guibas. A metric for distributions with applications to image databases. In Sixth International Conference on Computer Vision (IEEE Cat. No. 98CH36271), pages 59–66. IEEE, 1998.</p>
</div>
<div class="section level2">
<h2 id="examples-10">Examples<a class="anchor" aria-label="anchor" href="#examples-10"></a></h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/jaccard_run_emd.html">jaccard_run_emd</a></span><span class="op">(</span><span class="va">list</span>,<span class="st">"PEF"</span>,<span class="fl">10</span>,<span class="st">"hierarchical"</span>,<span class="fl">1</span>,<span class="fl">3</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.005</span>,<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="max_fluc"><code>max_fluc</code><a class="anchor" aria-label="anchor" href="#max_fluc"></a></h1>
<p>Determine the pair of maximum fluctuation difference on time series data distribution from a preprocessed list or Earth Mover’s Distance square matrix.</p>
<div class="section level2">
<h2 id="description-12">Description<a class="anchor" aria-label="anchor" href="#description-12"></a></h2>
<p>Determine the pair of maximum fluctuation difference on time series data distribution from a preprocessed list or Earth Mover’s Distance square matrix.</p>
</div>
<div class="section level2">
<h2 id="usage-10">Usage<a class="anchor" aria-label="anchor" href="#usage-10"></a></h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/max_fluc.html">max_fluc</a></span><span class="op">(</span><span class="va">input</span>, <span class="va">parameter</span>, <span class="va">maxIter</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-10">Arguments<a class="anchor" aria-label="anchor" href="#arguments-10"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>input</code></td>
<td>Either a list storing time series data or EMD martrix (also see functions: <a href="#patientlist">patient_list</a> , <a href="#emdmatrix">emd_matrix</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>Parameter of interest from time series data list</td>
</tr><tr class="odd"><td><code>maxIter</code></td>
<td>Maximum of iterations to apply for calculation of Earth Mover’s Distannce (also see function: <a href="#emdmatrix">emd_matrix</a> )</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-11">Value<a class="anchor" aria-label="anchor" href="#value-11"></a></h2>
<p>Console output with Patient_ID pair, corresponding Earth Mover’s Distance and visualized boxplot of both time series data distributions</p>
</div>
<div class="section level2">
<h2 id="examples-11">Examples<a class="anchor" aria-label="anchor" href="#examples-11"></a></h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="fu"><a href="reference/max_fluc.html">max_fluc</a></span><span class="op">(</span><span class="va">list</span>, <span class="st">"PEF"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="patient_boxplot"><code>patient_boxplot</code><a class="anchor" aria-label="anchor" href="#patient_boxplot"></a></h1>
<p>Visualize patient(s) time series from a preprocessed list with in a boxplot, either as normalized or non-normalized for an indicated parameter.</p>
<div class="section level2">
<h2 id="description-13">Description<a class="anchor" aria-label="anchor" href="#description-13"></a></h2>
<p>Visualize patient(s) time series from a preprocessed list with in a boxplot, either as normalized or non-normalized for an indicated parameter.</p>
</div>
<div class="section level2">
<h2 id="usage-11">Usage<a class="anchor" aria-label="anchor" href="#usage-11"></a></h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/patient_boxplot.html">patient_boxplot</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">patients</span>, <span class="va">parameter</span>, <span class="va">normalized</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-11">Arguments<a class="anchor" aria-label="anchor" href="#arguments-11"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>patients</code></td>
<td>Patient_ID(s) referring to (a) list element; can be single ID or multiple IDs (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="odd"><td><code>parameter</code></td>
<td>Parameter of interest in list element(s)</td>
</tr><tr class="even"><td><code>normalized</code></td>
<td>TRUE/FALSE if z-normalized (TRUE by default)</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-12">Value<a class="anchor" aria-label="anchor" href="#value-12"></a></h2>
<p>Visualized patient(s) time series data in a boxplot for indicated parameter</p>
</div>
<div class="section level2">
<h2 id="examples-12">Examples<a class="anchor" aria-label="anchor" href="#examples-12"></a></h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="fu"><a href="reference/patient_boxplot.html">patient_boxplot</a></span><span class="op">(</span><span class="va">list</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ID_2"</span>,<span class="st">"testpat_1"</span>,<span class="st">"testpat_2"</span>,<span class="st">"a301"</span><span class="op">)</span>, <span class="st">"FEV1"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="patient_hist"><code>patient_hist</code><a class="anchor" aria-label="anchor" href="#patient_hist"></a></h1>
<p>Visualize patient time series data from a preprocessed in a histogram for indicated parameter either normalized or non-normalized.</p>
<div class="section level2">
<h2 id="description-14">Description<a class="anchor" aria-label="anchor" href="#description-14"></a></h2>
<p>Visualize patient time series data from a preprocessed in a histogram for indicated parameter either normalized or non-normalized.</p>
</div>
<div class="section level2">
<h2 id="usage-12">Usage<a class="anchor" aria-label="anchor" href="#usage-12"></a></h2>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/patient_hist.html">patient_hist</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">Patient_ID</span>, <span class="va">parameter</span>, <span class="va">normalized</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-12">Arguments<a class="anchor" aria-label="anchor" href="#arguments-12"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>Patient_ID</code></td>
<td>Patient_ID referring to a list element (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="odd"><td><code>parameter</code></td>
<td>Parameter of interest in list element</td>
</tr><tr class="even"><td><code>normalized</code></td>
<td>TRUE/FALSE if z-normalized (TRUE by default)</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-13">Value<a class="anchor" aria-label="anchor" href="#value-13"></a></h2>
<p>Visualized patient time series data in a histogram for indicated parameter</p>
</div>
<div class="section level2">
<h2 id="examples-13">Examples<a class="anchor" aria-label="anchor" href="#examples-13"></a></h2>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span><span class="st">"
https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="fu"><a href="reference/patient_hist.html">patient_hist</a></span><span class="op">(</span><span class="va">list</span>,<span class="st">"testpat_1"</span>,<span class="st">"PEF"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="patient_list"><code>patient_list</code><a class="anchor" aria-label="anchor" href="#patient_list"></a></h1>
<p>Process patient time series data by interpolation options and store data in an object of type list.</p>
<div class="section level2">
<h2 id="description-15">Description<a class="anchor" aria-label="anchor" href="#description-15"></a></h2>
<p>Process patient time series data by interpolation options and store data in an object of type list.</p>
</div>
<div class="section level2">
<h2 id="usage-13">Usage<a class="anchor" aria-label="anchor" href="#usage-13"></a></h2>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">GitHub</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-13">Arguments<a class="anchor" aria-label="anchor" href="#arguments-13"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>path</code></td>
<td>Path where csv file(s) are stored (only folder, not specific file(s))</td>
</tr><tr class="even"><td><code>GitHub</code></td>
<td>Set TRUE when csv file comes form GitHub (FALSE by default); only in demo needed</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-9">Details<a class="anchor" aria-label="anchor" href="#details-9"></a></h2>
<p>Prior to undertaking an analysis using one of the FBC procedures, it is necessary to adequately process and prepare the relevant time series data. The function then creates an interactive flow using the console in R Studio. To begin, the method retrieves all csv files in the provided folder, indicating that it is capable of handling multiple files. The function extracts all csv files from the given directory and merges them into a single raw data frame. The user then indicates which column represents Patient ID and time for adequate processing. The csv files are merged, columns are selected where the Patient ID column will be renamed ”Patient_ID” and the time column will be titled ”Time”. This standardization approach is critical for subsequent features because it enables the easy detection of time series data and the consistent computation and processing of data, for example z-normalization.</p>
<p>The user should also indicate in the interactive console the time formate which will be standardized with the help of <a href="#lubridate">lubridate</a> . This is crucial because the technique can now filter the raw data by Patient ID, extract the start and end timestamps for each Patient ID, and then align the data if any records are missing while maintaining the indicated sample frequency.</p>
<p>The user may choose between seven approaches: L1 Regularization/Least absolute shrinkage and selection operator (LASSO) Regression, L2 Regularization/Ridge Regression, Elastic Net Regularization, Linear interpolation, Cubic C2 interpolation or, according to recent articles, fill in missing values using the highest or lowest quartile of measurements in the given time series data distribution.</p>
<p>The Regression and Regularization techniques generate adequate polynomials for each possible degree n-1 (where n is the total number of data points). Afterwards, cross-validation (from <a href="#glmnet">glmnet</a> ) is applied to determine the lambda value for the lowest MSE of the model. Afterwards, the model with polynomial degree for the lowest MSE is chosen and the missing data is interpolated with the regularized model.</p>
<p>It is also possible to apply a simple linear interpolation in between missing time series data points. It may be the easiest option to employ straight lines between neighboring points (also see <a href="#nainterpolation">na_interpolation</a> ). Nevertheless, these basic spline polynomials may be notoriously inexact. Cubic spline polynomials mostly provide better results.</p>
<p>Another option for the user is to apply the interpolation by using a cubic C spline. It implies that the composite function S must be twice continuously differentiable from all boundaries or subintervals (also see <a href="#nainterpolation">na_interpolation</a> ).</p>
<p>Without regression, regularization or interpolation, the user may opt to sample missing values within time series data by randomly choosing a value from the greatest or lowest quartile readings from each patient distribution. The R function then loops over each NA element in the time series data distribution of a patient for the specified parameter and randomly samples a value for the chosen quartile until the data frame is complete.</p>
</div>
<div class="section level2">
<h2 id="value-14">Value<a class="anchor" aria-label="anchor" href="#value-14"></a></h2>
<p>Object of type list storing patient time series data</p>
</div>
<div class="section level2">
<h2 id="references-7">References<a class="anchor" aria-label="anchor" href="#references-7"></a></h2>
<p>Jerome Friedman, Trevor Hastie, Rob Tibshirani, Balasubramanian Narasimhan, Ken- neth Tay, Noah Simon, and Junyang Qian. Package ‘glmnet’. Journal of Statistical Software. 2010a, 33(1), 2021.</p>
<p>Hui Zou and Trevor Hastie. Regularization and variable selection via the elastic net. Journal of the royal statistical society: series B (statistical methodology), 67(2):301– 320, 2005.</p>
<p>Steffen Moritz and Thomas Bartz-Beielstein. imputets: time series missing value imputation in r. R J., 9(1):207, 2017.</p>
</div>
<div class="section level2">
<h2 id="examples-14">Examples<a class="anchor" aria-label="anchor" href="#examples-14"></a></h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="patient_ts_plot"><code>patient_ts_plot</code><a class="anchor" aria-label="anchor" href="#patient_ts_plot"></a></h1>
<p>Visualize patient time series data from a preprocessed in a time series plot for an indicated parameter, either as normalized or non-normalized.</p>
<div class="section level2">
<h2 id="description-16">Description<a class="anchor" aria-label="anchor" href="#description-16"></a></h2>
<p>Visualize patient time series data from a preprocessed in a time series plot for an indicated parameter, either as normalized or non-normalized.</p>
</div>
<div class="section level2">
<h2 id="usage-14">Usage<a class="anchor" aria-label="anchor" href="#usage-14"></a></h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/patient_ts_plot.html">patient_ts_plot</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">Patient_ID</span>, <span class="va">parameter</span>, <span class="va">normalized</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-14">Arguments<a class="anchor" aria-label="anchor" href="#arguments-14"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>Patient_ID</code></td>
<td>Patient_ID referring to a list element (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="odd"><td><code>parameter</code></td>
<td>Parameter of interest in list element</td>
</tr><tr class="even"><td><code>normalized</code></td>
<td>TRUE/FALSE if z-normalized (TRUE by default)</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-15">Value<a class="anchor" aria-label="anchor" href="#value-15"></a></h2>
<p>Visualized patient time series data in a time series plot for indicated parameter</p>
</div>
<div class="section level2">
<h2 id="examples-15">Examples<a class="anchor" aria-label="anchor" href="#examples-15"></a></h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="fu"><a href="reference/patient_ts_plot.html">patient_ts_plot</a></span><span class="op">(</span><span class="va">list</span>,<span class="st">"testpat_1"</span>,<span class="st">"PEF"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="rnd_dat_rm"><code>rnd_dat_rm</code><a class="anchor" aria-label="anchor" href="#rnd_dat_rm"></a></h1>
<p>Remove a specific amount of data randomly from a time series data list.</p>
<div class="section level2">
<h2 id="description-17">Description<a class="anchor" aria-label="anchor" href="#description-17"></a></h2>
<p>Remove a specific amount of data randomly from a time series data list.</p>
</div>
<div class="section level2">
<h2 id="usage-15">Usage<a class="anchor" aria-label="anchor" href="#usage-15"></a></h2>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/rnd_dat_rm.html">rnd_dat_rm</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">removal</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-15">Arguments<a class="anchor" aria-label="anchor" href="#arguments-15"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>Object of type list storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>removal</code></td>
<td>Amount of data removal (0 = 0%, 1 = 100%)</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-16">Value<a class="anchor" aria-label="anchor" href="#value-16"></a></h2>
<p>Object of type list storing patient time series data with indicated amount of data removed randomly</p>
</div>
<div class="section level2">
<h2 id="examples-16">Examples<a class="anchor" aria-label="anchor" href="#examples-16"></a></h2>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">list_rm</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/rnd_dat_rm.html">rnd_dat_rm</a></span><span class="op">(</span><span class="va">testlist</span>, <span class="fl">0.95</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="sim_jaccard_cognate"><code>sim_jaccard_cognate</code><a class="anchor" aria-label="anchor" href="#sim_jaccard_cognate"></a></h1>
<p>Simulate random data removal for a removal amount with indicated number of simulations from time series data list and determine Jaccard index for all cluster via Cognate Cluster cluster assignment approach</p>
<div class="section level2">
<h2 id="description-18">Description<a class="anchor" aria-label="anchor" href="#description-18"></a></h2>
<p>Simulate random data removal for a removal amount with indicated number of simulations from time series data list and determine Jaccard index for all cluster via Cognate Cluster cluster assignment approach</p>
</div>
<div class="section level2">
<h2 id="usage-16">Usage<a class="anchor" aria-label="anchor" href="#usage-16"></a></h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">sim_jaccard_cognate</span><span class="op">(</span><span class="va">plist</span>, <span class="va">parameter</span>, <span class="va">removal</span>, <span class="va">n_simu</span>, <span class="va">method</span>, <span class="va">n_clust</span>, <span class="va">Iter</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-16">Arguments<a class="anchor" aria-label="anchor" href="#arguments-16"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>Object of type list storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>Parameter of interest in time series data list</td>
</tr><tr class="odd"><td><code>removal</code></td>
<td>Amount of random data removal to determine Jaccard index</td>
</tr><tr class="even"><td><code>n_simu</code></td>
<td>Number of simulations</td>
</tr><tr class="odd"><td><code>method</code></td>
<td>Clustering method (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="even"><td><code>n_clust</code></td>
<td>Number of clusters (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="odd"><td><code>Iter</code></td>
<td>Maximum iterations to determine Earth Mover’s Distances (also see function: <a href="#emdmatrix">emd_matrix</a> ); default is 5,000 for this function</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-10">Details<a class="anchor" aria-label="anchor" href="#details-10"></a></h2>
<p>The cognate cluster approach works in the manner that first a Gold Standard cluster is determined meaning the cluster assignments without any data removal. Subsequently, random data is removed from the original, complete data and clustering is performed again on the leaky data. The cluster determined to be cognate to the Gold Standard cluster is the one with the highest overlap in cluster members, meaning hte cluster with highest acheived Jaccard index. Afterwards, the Jaccard indices are calculated, comparing cluster members with complete and leaky data, for each cluster.</p>
</div>
<div class="section level2">
<h2 id="value-17">Value<a class="anchor" aria-label="anchor" href="#value-17"></a></h2>
<p>Object of type matrix storing received Jaccard indices for indicated amount of random data removal for all clusters</p>
</div>
<div class="section level2">
<h2 id="references-8">References<a class="anchor" aria-label="anchor" href="#references-8"></a></h2>
<p>Anja Jochmann, Luca Artusio, Hoda Sharifian, Angela Jamalzadeh, Louise J Fleming, Andrew Bush, Urs Frey, and Edgar Delgado-Eckert. Fluctuation-based clustering reveals phenotypes of patients with different asthma severity. ERJ open research, 6(2), 2020.</p>
</div>
<div class="section level2">
<h2 id="examples-17">Examples<a class="anchor" aria-label="anchor" href="#examples-17"></a></h2>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#Sampling frequency is supposed to be daily</span>
<span class="va">output</span> <span class="op">&lt;-</span> <span class="fu">sim_jaccard_cognate</span><span class="op">(</span><span class="va">list</span>, <span class="st">"PEF"</span>, <span class="fl">0.05</span>, <span class="fl">10</span>, <span class="st">"hierarchical"</span>, <span class="fl">2</span>, <span class="fl">1000</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="sim_jaccard_emd"><code>sim_jaccard_emd</code><a class="anchor" aria-label="anchor" href="#sim_jaccard_emd"></a></h1>
<p>Simulate random data removal for a removal amount with indicated number of simulations from time series data list and determine Jaccard index for all clusters via Earth Mover’s distance cluster assignment approach.</p>
<div class="section level2">
<h2 id="description-19">Description<a class="anchor" aria-label="anchor" href="#description-19"></a></h2>
<p>Simulate random data removal for a removal amount with indicated number of simulations from time series data list and determine Jaccard index for all clusters via Earth Mover’s distance cluster assignment approach.</p>
</div>
<div class="section level2">
<h2 id="usage-17">Usage<a class="anchor" aria-label="anchor" href="#usage-17"></a></h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/sim_jaccard_emd.html">sim_jaccard_emd</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">parameter</span>, <span class="va">removal</span>, <span class="va">n_simu</span>, <span class="va">method</span>, <span class="va">n_clust</span>, <span class="va">Iter</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-17">Arguments<a class="anchor" aria-label="anchor" href="#arguments-17"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>Object of type list storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>parameter</code></td>
<td>Parameter of interest in time series data list</td>
</tr><tr class="odd"><td><code>removal</code></td>
<td>Amount of random data removal to determine Jaccard index</td>
</tr><tr class="even"><td><code>n_simu</code></td>
<td>Number of simulations</td>
</tr><tr class="odd"><td><code>method</code></td>
<td>Clustering method (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="even"><td><code>n_clust</code></td>
<td>Number of clusters (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="odd"><td><code>Iter</code></td>
<td>Maximum iterations to determine Earth Mover’s Distances (also see function: <a href="#emdmatrix">emd_matrix</a> ); default is 5,000 for this function</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-11">Details<a class="anchor" aria-label="anchor" href="#details-11"></a></h2>
<p>This method represents a novel approach and potential complementary method to <a href="#simjaccardcognate">sim_jaccard_cognate</a> . First, clustering is performed on complete data without removal, servinng as Gold Standard clusters. For every Gold Standard cluster then, all time series data from all patients is z-normalized and then assumed to be as one Gold Standard distribution. Subsequently, random data is removed form the time series data. Each leaky data distribution is then compared via Earth Mover’s Distance to each Gold Standard Distribution. The Gold Standard cluster distribution to which the observed leaky distribution exhibits the lowest Earth Mover’s Distance gets the assignment. This process is repeated until every leaky time series data distribution is assigned to a cluster. Afterwards, the Jaccard indices are calculated, comparing cluster members with complete and leaky data, for each cluster.</p>
</div>
<div class="section level2">
<h2 id="value-18">Value<a class="anchor" aria-label="anchor" href="#value-18"></a></h2>
<p>Object of type matrix storing received Jaccard indices for indicated amount of random data removal for all clusters</p>
</div>
<div class="section level2">
<h2 id="references-9">References<a class="anchor" aria-label="anchor" href="#references-9"></a></h2>
<p>Yossi Rubner, Carlo Tomasi, and Leonidas J Guibas. A metric for distributions with applications to image databases. In Sixth International Conference on Computer Vision (IEEE Cat. No. 98CH36271), pages 59–66. IEEE, 1998.</p>
</div>
<div class="section level2">
<h2 id="examples-18">Examples<a class="anchor" aria-label="anchor" href="#examples-18"></a></h2>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/patient_list.html">patient_list</a></span><span class="op">(</span>
<span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,
GitHub <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/sim_jaccard_emd.html">sim_jaccard_emd</a></span><span class="op">(</span><span class="va">list</span>, <span class="st">"PEF"</span>, <span class="fl">0.05</span>, <span class="fl">10</span>, <span class="st">"hierarchical"</span>, <span class="fl">2</span>, <span class="fl">100</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="sim_sample_enr"><code>sim_sample_enr</code><a class="anchor" aria-label="anchor" href="#sim_sample_enr"></a></h1>
<p>Simulate random sampling for NA entries in enrichment data and check stability of resulting p-values for the enrichment parameters for an indicated number of random sampling simulations.</p>
<div class="section level2">
<h2 id="description-20">Description<a class="anchor" aria-label="anchor" href="#description-20"></a></h2>
<p>Simulate random sampling for NA entries in enrichment data and check stability of resulting p-values for the enrichment parameters for an indicated number of random sampling simulations.</p>
</div>
<div class="section level2">
<h2 id="usage-18">Usage<a class="anchor" aria-label="anchor" href="#usage-18"></a></h2>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/sim_sample_enr.html">sim_sample_enr</a></span><span class="op">(</span><span class="va">plist</span>, <span class="va">path</span>, <span class="va">clustdat</span>, <span class="va">clustno</span>, <span class="va">n_sim</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-18">Arguments<a class="anchor" aria-label="anchor" href="#arguments-18"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>plist</code></td>
<td>List storing patient time series data (also see function: <a href="#patientlist">patient_list</a> )</td>
</tr><tr class="even"><td><code>path</code></td>
<td>Path where enrichment csv file is stored</td>
</tr><tr class="odd"><td><code>clustdat</code></td>
<td>Object of type list storing clustering data (also see function: <a href="#clustmatrix">clust_matrix</a> )</td>
</tr><tr class="even"><td><code>clustno</code></td>
<td>Cluster number of interest</td>
</tr><tr class="odd"><td><code>n_sim</code></td>
<td>Number of simulations</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="details-12">Details<a class="anchor" aria-label="anchor" href="#details-12"></a></h2>
<p>It allows the sampling in NA entries to be repeated for each parameter in the enriched data set. The primary objective here is to validate the random sampling process for missing data by running many simulations and comparing the resultant p-values. An enrichment data frame with NA elements is saved as a simulation foundation. This data frame will always serve as the foundation for any subsequent simulations added. Following that, the program runs through each NA item in the dataset and generates a random sample of the current parameter’s distribution. After completing this step for each parameter, the function generates the associated p-values as explained in <a href="#enrobsclust">enr_obs_clust</a> .</p>
</div>
<div class="section level2">
<h2 id="value-19">Value<a class="anchor" aria-label="anchor" href="#value-19"></a></h2>
<p>Object of type list storing the received p-values for each parameter in a vector and boxplot visualizing the received p-values</p>
</div>
<div class="section level2">
<h2 id="examples-19">Examples<a class="anchor" aria-label="anchor" href="#examples-19"></a></h2>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>list <span class="ot">&lt;-</span> <span class="fu">patient_list</span>(</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="st">"https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/phys/data.csv"</span>,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="at">GitHub =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Sampling frequency is supposed to be daily</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">'https://raw.githubusercontent.com/MrMaximumMax/FBCanalysis/master/demo/enrich/enrichment.csv'</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> <span class="fu">sim_sample_enr</span>(list,path,clustering,<span class="dv">1</span>,<span class="dv">100</span>)</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>sim_sample_enr <span class="ot">&lt;-</span> <span class="cf">function</span>(plist, path, clustdat, clustno, n_sim)</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="znorm"><code>znorm</code><a class="anchor" aria-label="anchor" href="#znorm"></a></h1>
<p>z-normalise data</p>
<div class="section level2">
<h2 id="description-21">Description<a class="anchor" aria-label="anchor" href="#description-21"></a></h2>
<p>Function applicable on any numeric distribution of data and z-normalize them.</p>
</div>
<div class="section level2">
<h2 id="usage-19">Usage<a class="anchor" aria-label="anchor" href="#usage-19"></a></h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/znorm.html">znorm</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="arguments-19">Arguments<a class="anchor" aria-label="anchor" href="#arguments-19"></a></h2>
<table class="table"><colgroup><col width="44%"><col width="55%"></colgroup><thead><tr class="header"><th>Argument</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td><code>data</code></td>
<td>Numeric distribution; may be stored in an object of type vector, matrix or data frame</td>
</tr></tbody></table></div>
<div class="section level2">
<h2 id="value-20">Value<a class="anchor" aria-label="anchor" href="#value-20"></a></h2>
<p>z-normalized distribution of input</p>
</div>
<div class="section level2">
<h2 id="examples-20">Examples<a class="anchor" aria-label="anchor" href="#examples-20"></a></h2>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">random_distribution</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">50</span>, min <span class="op">=</span> <span class="fl">1</span>, max <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="va">znorm_distribution</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/znorm.html">znorm</a></span><span class="op">(</span><span class="va">random_distribution</span><span class="op">)</span></code></pre></div>
</div>
</div>


  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Maximilian Koehler.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

